clients/js/src/ChromaClient.ts

```diff
@@ -78,10 +104,12 @@ export class ChromaClient {
       database: database,
     });
 
-    // TODO: Validate tenant and database on client creation
```

Comment:
interesting, wondering if it's worth wrapping the class with a Proxy so we don't have to manually add it to each public method?

Reply:
Yeah that is a good callout. I think there's a high likelihood that this will get dropped in the future. One thing that the current approach allows is to put certain validations before the `await initPromise;` e.g. 

```TypeScript
if (typeof params.ids === "string") {
  throw new Error(
    "To add a single record, use the singular field names (id, embedding, metadata, document) instead of the plural field names (ids, embeddings, metadatas, documents)",
  );
}
await this.initPromise;
```

Also, ideally, this would only apply to async functions. Do you know if there is a way to detect that in a proxy and only wrap the async functions? Obviously we only have async functions today but I would worry about accidentally making a synchronous function async.

Reply:
But yeah not opposed to this. My gut reaction would be to hold off on this, but happy to do this in this PR if you think that's the better trade-off.

Reply:
I think we can leave as-is for now, easy to adjust later